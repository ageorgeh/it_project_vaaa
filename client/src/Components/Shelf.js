import React from 'react'

class Shelf extends React.Component {
  constructor (props) {
    super(props)
    this.selectShelf = this.selectShelf.bind(this)
  }

  selectShelf () {
    this.props.onSelect(this.props.shelfKey)
  }

  render () {
    return (
            <>
                <li>
                    <div className="mx-2">
                        <button onClick={this.selectShelf} className="flex items-center py-2 w-full font-normal text-stone-900 rounded-lg dark:text-white hover:bg-stone-100 pl-3 dark:hover:bg-stone-700">
                        {this.props.name}
                        </button>
                    </div>
                </li>
            </>
    )
  }
}

export default Shelf
