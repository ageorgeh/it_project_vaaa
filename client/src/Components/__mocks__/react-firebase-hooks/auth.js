import { jest } from '@jest/globals'
export const useAuthState = jest.fn()
