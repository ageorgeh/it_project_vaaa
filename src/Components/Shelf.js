function Shelf(props) {
    return (
    <>
    <div className="text-center">
        <button type="button" className="text-white text-center rounded-lg bg-gray-800 hover:bg-gray-900 font-medium text-sm px-5 py-2.5 mr-2 mb-2">{props.name}</button>
    </div>
    </>
  );
}

export default Shelf;