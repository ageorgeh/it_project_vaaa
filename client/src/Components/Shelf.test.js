/* eslint-disable no-unused-vars */
import React from 'react'
import { render, screen } from '@testing-library/react'
import Shelf from './Shelf'
import { BrowserRouter } from 'react-router-dom'
import '@testing-library/jest-dom'
import { useAuthState } from 'react-firebase-hooks/auth'

const shelves = [
  { name: 'All Books', shelfID: 'All Books' },
  { name: 'Test', shelfID: 'Test' }
]

useAuthState.mockReturnValue([false, false])

test.each(shelves)(
  'Check Book has the current book: %s',
  (shelf) => {
    render(
      <BrowserRouter>
      <Shelf
        name={shelf.name}
        key={0}
        onSelect={null}
        shelfID={shelf.shelfID}
        onDeleteClose={null}
    />
      </BrowserRouter>
    )
    const title = screen.getByText(shelf.name)
    expect(title).not.toBeNull()
  }
)
