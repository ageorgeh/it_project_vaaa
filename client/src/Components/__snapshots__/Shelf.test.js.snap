// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays shelf 1`] = `
<li>
  <div
    className="mx-2"
  >
    <button
      className="flex items-center py-2 w-full font-normal text-stone-900 rounded-lg dark:text-white hover:bg-stone-100 pl-3 dark:hover:bg-stone-700"
      onClick={[Function]}
    />
  </div>
</li>
`;
